client/components/jetpack/card/jetpack-free-card-button/index.tsx

```diff
@@ -38,12 +38,17 @@ const JetpackFreeCardButton: FC< JetpackFreeCardButtonProps > = ( {
 } ) => {
 	const translate = useTranslate();
 	const wpAdminUrl = useSelector( getJetpackWpAdminUrl );
+	const { site } = urlQueryArgs;
 
 	// If the user is not logged in and there is a site in the URL, we need to construct
 	// the URL to wp-admin from the `site` query parameter
-	const wpAdminUrlFromQuery = urlQueryArgs.site
+	const wpAdminUrlFromQuery = site
 		? getUrlFromParts( {
-				...getUrlParts( `https://${ urlQueryArgs.site }/wp-admin/admin.php` ),
+				...getUrlParts(
+					`https://${
+						isJetpackSecondarySiteSlug( site ) ? site.replace( '::', '/' ) : site
```

Comment:
Does this code need this function? From what I can tell, it uses regex to check for `::` only to replace it with `/`. Would simply `site.replace( '::', '/' )` work?

Reply:
Most likely, but the check seemed safer if `::` was somehow used differently.

Reply:
Perhaps, but maybe doing a replace and validating via `URL` would be better? I'm concerned there'll be domain names that don't fit a hard-coded regex, such as i18n domain names.

Reply:
That's fair. I updated the PR.